File: insertion_sort.cpp
#include <iostream>

using namespace std;

int main(void) {
    int arr[10] = {9, 0, 8, 1, 7, 2, 6, 3, 5, 4};

    int key, j;
    for(int i=1; i<10; i++) {
        key = arr[i];
        j = i - 1;

        while(j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j];
            j -= 1;
        }
        arr[j+1] = key;
    }

    for(int k=0; k<10; k++) {
        cout << *(arr + k) << " ";
    }

    cout << endl;
    return 0;
}

File: golang/insertion_sort.go
package main

import "fmt"

func main() {
	arr := [10]int{9, 0, 1, 8, 2, 7, 3, 6, 4, 5}

	for i := 1; i < len(arr); i++ {
		j := i

		for j > 0 {
			if arr[j-1] > arr[j] {
				arr[j-1], arr[j] = arr[j], arr[j-1]
			}

			j -= 1
		}
	}
	fmt.Println(arr)
}
